src/codegen/mod.rs

```diff
@@ -2172,6 +2177,206 @@ impl CodeGenerator for CompInfo {
+            ("operator^=", "BitXorAssign", "bitxor_assign"),
+            ("operator/=", "DivAssign", "div_assign"),
+            ("operator%=", "RemAssign", "rem_assign"),
+            ("operator<<=", "ShlAssign", "shl_assign"),
+            ("operator>>=", "ShrAssign", "shr_assign"),
+            ("operator-=", "SubAssign", "sub_assign"),
+        ];
+        let binary_operators = [
+            ("operator+", "Add", "add"),
+            ("operator&", "BitAnd", "bitand"),
+            ("operator|", "BitOr", "bitor"),
+            ("operator^", "BitXor", "bitxor"),
+            ("operator/", "Div", "div"),
+            ("operator%", "Rem", "rem"),
+            ("operator<<", "Shl", "shl"),
+            ("operator>>", "Shr", "shr"),
+            ("operator-", "Sub", "sub"),
+        ];
+        let unary_operators = [
+            ("operator-", "Neg", "neg"),
+            ("operator!", "Not", "not"),
+        ];
+        if name == "operator==" {
+            result.push(quote!(
+                impl PartialEq for #ty_for_impl {
+                    fn eq(&self, rhs: &Self) -> bool {
+                        unsafe {
+                            #function_name(self, rhs)
+                        }
+                    }
+                }
+            ));
+            return;
+        }
+        for el in assignment_operators.iter() {
+            // The args.len() == 2 check shouldn't be neccessary but we still have it.
+            if args.len() == 2 && name == el.0 {
+                let rhs_type = second_arg_type.unwrap();
+                let trait_name = proc_macro2::TokenStream::from_str(el.1).unwrap();
+                let func_name = proc_macro2::TokenStream::from_str(el.2).unwrap();
+                result.push(quote!(
+                    impl std::ops::#trait_name<#rhs_type> for #ty_for_impl {
+                        fn #func_name(&mut self, rhs: #rhs_type) {
+                            let retptr = unsafe {
+                                #function_name(self, rhs)
+                            };
+                            assert_eq!(retptr, self as *mut #ty_for_impl, "The bindgen authors have no idea how to handle this case.");
+                        }
+                    }
+                ));
+                return;
+            }
+        }
+        for el in binary_operators.iter() {
+            // The args.len() == 2 check is needed, because "operator-" is both
+            // the name of the binary minus and the unary minus operator (a.k.a
+            // the negation operator).
+            if args.len() == 2 && name == el.0 {
+                let rhs_type = second_arg_type.unwrap();
+                let trait_name = proc_macro2::TokenStream::from_str(el.1).unwrap();
+                let func_name = proc_macro2::TokenStream::from_str(el.2).unwrap();
+                result.push(quote!(
+                    impl std::ops::#trait_name<#rhs_type> for &#ty_for_impl {
```

Comment:
This needs to preserve the trait_prefix, to handle `use_core` correctly.

Reply:
Fixed in the next commit